vault/token_store.go

```diff
@@ -917,23 +1069,187 @@ func (ts *TokenStore) destroyCubbyhole(saltedID string) error {
 
 func (ts *TokenStore) authRenew(
 	req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
+	if req.Auth == nil {
+		return nil, fmt.Errorf("request auth is nil")
+	}
+
+	f := framework.LeaseExtend(req.Auth.Increment, 0, ts.System())
+
+	idInt, ok := req.Auth.InternalData["id"]
```

Comment:
I'll check this, but I don't think it's correct. TokenStore's `handleRenew` calls out to Expiration's `RenewToken`. This loads the lease, then calls `renewAuthEntry`, which does an explicit `auth.ClientToken=""` before passing the auth info over to Logical's `RenewAuthRequest`, which eventually feeds into this function.